astro/src/content/docs/operate/deploy/terraform.mdx

```diff
@@ -0,0 +1,432 @@
+* Pasting the copied HCL into that file.
+* Adding the previously created variable as `var.fusionauth_api_key` to the `provider` section
+* Adding the FusionAuth instance you plan to manage with the `var.fusionauth_host` variable in the `provider` section
+
+Here's a screenshot of the page.
+
+![Downloading the HCL for the FusionAuth provider.](/img/docs/operate/deploy/terraform-provider.png)
+
+Here's what your `main.tf` file should look like at this point.
+
+<RemoteCode lang="hcl" title="Example main.tf File" url="https://raw.githubusercontent.com/fusionauth/fusionauth-example-terraform/main/examples/import/main.tf" tags="terraformProvider" />
+
+Finally, the following command prepares the current working directory for use with Terraform.
+
+```shell
+terraform init
+```
+
+It is always safe to run [`terraform init`](https://developer.hashicorp.com/terraform/cli/commands/init) multiple times, to bring the working directory up to date with changes in the configuration. Though subsequent runs may give errors, this command will never delete your existing configuration or state.
+
+## Choosing A Configuration Strategy
+
+Once you initialize the Terraform working directory, review different initial strategies to handle resource creation.
+
+| Strategy | Advantages | Disadvantages |
+| ---- | ---- | ---- |
+| import | manage existing resources | some resources don't fully support the Terraform lifecycle |
+| data source  | reference existing resources without having to manage them | requires a external process to manage these resources |
+| create | always create and be able to manage all resources through Terraform | bigger initial effort and default resources can't be managed |
+
+## Importing Default Resources
+
+Before you create, update, and remove resources with Terraform, let's examine the [Import](https://developer.hashicorp.com/terraform/language/import) and [Data Source](https://developer.hashicorp.com/terraform/language/data-sources) functionality.
+
+There are [FusionAuth default configuration elements](/docs/get-started/core-concepts/limitations#default-configuration) present in every FusionAuth instance. If you want to manage changes to these elements via Terraform, you must tell Terraform about them. Or, if you have existing FusionAuth configuration you want to manage via Terraform, import it.
+
+<Aside type="tip">
+The most consistent method for managing FusionAuth is to create all resources through Terraform. However, you must use either `Import` or `Data Source` to handle default or existing FusionAuth configuration.
+</Aside>
+
+The FusionAuth default tenant and default Application are two configuration elements that you will almost certainly want to manage via Terraform. There are others outlined in the [FusionAuth default configuration elements](/docs/get-started/core-concepts/limitations#default-configuration) that can be managed using the same methods described below.
```

Comment:
With the new designs, are we still capitalizing FA object type names? I see Application and application both used in this doc.

Reply:
We are.

From the docsdevreadme:

```
Capitalize all domain objects, especially when working the object's API in which it is created and updated in FusionAuth. For example, see the API Key APIs description for apiKeyId, where API Key is capitalized: The unique Id of the API Key to create. If not specified a secure random UUID will be generated.
If referring to something that exists as a domain object in FusionAuth, but you are not explicitly referring to an object being created/updated in FusionAuth, use lowercase. Here are some examples: To allow users to log into and use your application, you’ll need to create an Application in FusionAuth.
```

I'll take another pass.